linked_list/implementation.py

```diff
@@ -7,34 +8,138 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
-
+        self.start = None # initialize starting node
+        self.end = None # initialize ending node
+        
+        if elements: # if elements is not none
+            for e in elements: # loop through each element
+                node = Node(e) # convert it into a node
+                if not self.start: # if self.start is none
+                    self.start = node # assign the node to it
+                    #print(self.start)
+                if self.end: # if self.end is not none
+                    self.end.next = node # assign node to its 'next' pointer
+                    #print(self.end)
+                self.end = node # assign the node to self.end, since we want self.end to be the last e we looped through
+        
     def __str__(self):
-        pass
+        return '{}'.format([i for i in self])
```

Comment:
Elegant solution, however to simplify it a little you could just use `return str([i for i in self])` instead of having to use format.